mkosi/burn.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+
+from mkosi.config import (
+    MkosiArgs,
+    MkosiConfig,
+    OutputFormat,
+)
+from mkosi.log import die
+from mkosi.run import run
+
+def run_burn(args: MkosiArgs, config: MkosiConfig) -> None:
+
+    if config.output_format not in (
+        OutputFormat.disk,
+        OutputFormat.esp,
+    ):
+        die(f"{config.output_format} images cannot be dd'ed to disk")
+
+    fname = config.output_dir_or_cwd() / config.output
+
+    if (len(args.cmdline) != 1):
+            die("Expected device argument.");
+
+    device = args.cmdline[0]
```

Comment:
Should we maybe do some sanity check before passing this on? e.g. wrapping this in a ``Path()` and then doing a `device.is_block_device()` check?

Reply:
a regular file is fine too.

repart does its own checking really, no need to replicate this here.